Added custom webkit scrollbars and settings for scrollbars

JadenTepper opened the pull request:
Hi! Here's a second PR about custom scrollbars.

## Motivation
On Windows machines, scrollbars are wide and can cover a bit of the rightmost column on a flow. In addition, they don't fit with the dark theme. 

## Changes
1) I added custom webkit scrollbars that support thinner and more thematic styles. 
**Note**: I added logic to keep the splash page using default-style scrollbars. However, I did change the overflow rules on the splash page to prevent an x-direction scrollbar from showing up on Windows machines.
2) I added a setting to control the width of the scrollbars as well as a toggle to disable them. 
**Note**: I did not know where to place the settings, so I made a new section. Also, I set the default to having the custom scrollbars on with 6px width (my preference), but feel free to change these. 
3) I changed overflow rules in certain locations to prevent poor scrolling behavior. Previously, in certain cases on Windows and maybe Mac, the nodes in a flow could be scrolled horizontally and become unaligned with the background columns.

## Testing
I hand-tested the scrollbars on Windows Chrome, Windows Edge, Windows Firefox, Mac Chrome, and Mac Safari. All scrollbars seem to be properly replaced, and the new overflow rules seem to prevent poor scrolling behavior without cutting off content. However, as before, the tests were not rigorous, so small problems may have been missed. 

Old browsers will likely fall back to legacy behavior, but their behavior is untested.

The new overflow rules could affect mobile, but this is also untested (I assume very few people are flowing on their phones, though iPads could be a more common use case).

---
As always, let me know what you think. I am happy to edit any changes to your suggestion.

Ashwagandhae commented:
Hi, thank you for the PR! I didn't know that scrollbars behaved that way, thank you for letting me know and creating a fix! It looks mostly good, I just have a few questions and suggestions. 

## Questions
- I see `overflow-y: clip` and `overflow-x: clip` added in a few places, why? It doesn't change much for me, besides preventing a "bouncing" animation when I scroll up and down passed the limit (on macOS). That bouncing seems to be native macOS scrolling behaviour, which I don't think should be tampered with. 
- Why does `src/routes/app/+page.svelte` have code that adds and removes an `app` class from the body? 

## Changes
- Make the setting off by default:
    - to ensure that it doesn't break anything for current users. 
    - because the auto-hide scrollbar setting, which I think is the default on most operating systems, bypasses the need for a custom scrollbar
- Change setting name `Scrollbar Width` to `Custom scrollbar width` 
- Reverse the order of the two scrollbar settings, users probably want to turn on custom scrollbar before changing its width
- Move the `Scrollbars` settings group between `Toolbar` and `Controls`, as it seems more related to those

Thank you again for taking the time to make this PR, and going through the effort to understand my incomprehensible settings code! I'll look at your other PR soon, I've been a bit busy with school.

JadenTepper commented:
**Preface**: The other PR, imho, is quite a bit more important than this one and should be faster to review (it's like 10 changed lines, and all changes are very simple), so if you don't have much time feel free to put this one on the back burner.

I updated the settings as suggested. 

As for the overflow style changes, MacOS on prod has no issues, but Windows 10 and Windows 11 machines have some problems (at least for the one Windows 10 machine and one Windows 11 machine I tested this on). Below is an image showing scrolling behavior on Windows. You can see a few things: 
1) The dark mode theming problem I was mentioning. 
2) Scrollbars on Windows have a non-zero width and don't disappear. As a result, when a vertical scroll bar is created in `Flow`, the width of `Flow` is slightly smaller than the CSS calculates, and a horizontal scrollbar is created. If that horizontal scrollbar gets scrolled, it shifts the `Box` nodes, as seen here.
3) Same thing happens with the scrollbars on `body`. If the page becomes thin enough horizontally that a scrollbar needs to be created, it makes a vertical scrollbar appear for that tiny sliver of the page covered by the horizontal scrollbar. The annoying part here is now that both scrollbars exist, they prevent each other from disappearing for the same reason even if the page is enlarged again.
![scroll example](https://github.com/user-attachments/assets/741aa127-3d2e-476b-954f-f9ba8ceaa338)

I added the clipping to various locations to prevent this behavior. On Mac, it may cause some different bouncing behavior when over-scrolling, but I felt that fixing this problem on non-Mac platforms may be worth the tradeoff. This is why I applied the new overflow settings without a setting to disable them. Let me know what you think.

I also added the .app class to the body through JS so that the vertical clip required on the body in  `/app`  route doesn't get applied to the splash page. There may be other ways to change the global clipping settings on `body` when just on the `/app` route, but I thought this was easiest.

Thank you for reviewing the request, and let me know if anything else needs changing.
